### PR TITLE
feat: print more phase ASTs

### LINKED ISSUE #5898: Re-tinker system to have a phase-based API and not AST-based
Right now we phrase the pretty printing as "pretty print the lifted ast" but that could be multiple phases. This ticket would
1. Rephrase the input string (e.g. "Typed Ast" -> "Typer")
2. Rename functions and comments internally
3. Rename the asts (e.g. cachedLiftedAst -> cachedClosureConvAst)
4. Possibly add more ast caching

### PR DESCRIPTION
**Pull request by JonathanStarup**
* change `--Xprint-asts` to `--Xprint-phase-asts`
* Rename cached ASTs to be named based on their phase, not their AST
* Caches all backend phases
* Add pretty printing for all supported phase asts

Fixes #5898

### CODE REVIEW

**Comment by magnus-madsen** [comment]
We need to check that this does not impact compile time. Can you do an experiment before/after with `--Xbenchmark-throughput`. Thanks!

File: main/src/ca/uwaterloo/flix/api/Flix.scala [hunk 1]
```diff
@@ -557,21 +605,24 @@ class Flix {
...
-    val afterTailrec = Tailrec.run(cachedLiftedAst)
-    val afterOptimizer = Optimizer.run(afterTailrec)
-    val afterLateTreeShaker = LateTreeShaker.run(afterOptimizer)
-    val afterReducer = Reducer.run(afterLateTreeShaker)
-    val afterVarNumbering = VarNumbering.run(afterReducer)
-    val afterMonoTyper = MonoTyper.run(afterVarNumbering)
-    cachedErasedAst = Eraser.run(afterMonoTyper)
-    val afterJvmBackend = JvmBackend.run(cachedErasedAst)
+    // clear caches to avoid asts from previous compilations on crashes
+    clearBackendCaches()
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/Flix.scala:609]: I understand why you do it, but it seems like a brittle hack :/

**Comment by JonathanStarup** [comment]
Master

Throughput (best): 10.630 lines/sec (with 8 threads.)

  min:  3.220, max: 10.630, avg:  8.376, median:  9.707

  The highest throughput was in iteration: 9 (out of 10).
  The ratio between the best and worst iteration was: 3,3x.

Finished 10 iterations on 64.172 lines of code in 86 seconds.

**Comment by JonathanStarup** [comment]
@magnus-madsen heres the benchmarks, looks like it matters a bit. 13.7 % decrease on median

**Comment by magnus-madsen** [comment]
let us see

**Pull request merged**